sormas-api/src/main/java/de/symeda/sormas/api/contact/ContactCriteria.java

```diff
@@ -464,4 +468,31 @@ public YesNoUnknown getReturningTraveler() {
 	public void setReturningTraveler(YesNoUnknown returningTraveler) {
 		this.returningTraveler = returningTraveler;
 	}
+
+	public void setEventLike(String eventLike) {
+		this.eventLike = eventLike;
+	}
+
+	public String getEventLike() {
```

Comment:
Add `@IgnoreForUrl` here as well to prevent personal data being leaked to the URL

Reply:
Unfortunately the link to contacts already filtered by events (#3034) won't work with this annotation

Reply:
I'd suggest a workaround then; add another property to the ContactCriteria that is simply called "eventUuidLike" and use that if coming from a contact; since this field will not be visible in the UI, users can't accidentally leak personal data with it

Reply:
That would do it, but it would be a bit wobbly in terms of user experience. 
Since there's no access to this field in the UI, you can't really know that this filter is applied without peeking at the URL and you're not able to turn it off directly from within the UI. You'd have to apply another filter, and only then use the `Reset filters` button.

I already made the changes, but this behaviour is bugging me a little bit. If you still think this is okay, i'll push it as is

Reply:
You're right - that will probably lead to confusion. Still I think it's not okay to go with the approach of just putting the information into the URL, so we need to find some kind of solution.

A VERY hacky approach could be to automatically also apply the UUID to the filter field that is visible to the user. So the UUID filter would actually be applied twice, but that should be fine. Clicking on "Reset filters" would then also get rid of the "hidden" filter applied by clicking the button in the event. Do you think that makes sense?

Reply:
We could add :

    if (criteria.getEventUuid() != null) {
	    criteria.eventLike(criteria.getEventUuid());
	    criteria.eventUuid(null);
    }

In the enter method of the contact view. It is indeed hacky, but it works.

It's either that or we add another field to filter on the UUID but that would be tantamount to the newly added eventLike field, and rather useless to the UX.

So, do I just go for the "hacky" way ?

Reply:
I'm fine with approaches that are a bit hacky. Vaadin won't be our UI framework forever anyway. :)

Your suggestion would work for me!

Reply:
Done !